### PR TITLE
fix: golangci-lint error

### PR DESCRIPTION
**Pull request by FrankYang0529**
#### Which issue(s) this PR fixes:
https://github.com/longhorn/longhorn/issues/7448

### CODE REVIEW

File: backupbackingimage/backupbackingimage.go [hunk 1]
```diff
@@ -298,93 +306,105 @@
...
 		return err
 	}
 
-	defer lock.Unlock()
 	if err := lock.Lock(); err != nil {
 		return err
 	}
+	defer func() {
+		if unlockErr := lock.Unlock(); unlockErr != nil {
+			logrus.WithError(unlockErr).Warn("Failed to unlock restore")
```
Comment (innobead) [inline backupbackingimage/backupbackingimage.go:338]: Suggest log: `Failed to unlock restore backing image` 

Please correct other logs if needed.

File: deltablock.go [hunk 2]
```diff
@@ -258,22 +268,38 @@ func CreateDeltaBlockBackup(backupName string, config *DeltaBackupConfig) (isInc
...
 		if progress, backup, err := performBackup(bsDriver, config, delta, deltaBackup, backupRequest.lastBackup); err != nil {
 			logrus.WithError(err).Errorf("Failed to perform backup for volume %v snapshot %v", volume.Name, snapshot.Name)
-			deltaOps.UpdateBackupStatus(snapshot.Name, volume.Name, string(types.ProgressStateInProgress), progress, "", err.Error())
+			if updateErr := deltaOps.UpdateBackupStatus(snapshot.Name, volume.Name, string(types.ProgressStateInProgress), progress, "", err.Error()); updateErr != nil {
+				logrus.WithError(updateErr).Error("Failed to update backup status")
+			}
 		} else {
-			deltaOps.UpdateBackupStatus(snapshot.Name, volume.Name, string(types.ProgressStateInProgress), progress, backup, "")
+			if updateErr := deltaOps.UpdateBackupStatus(snapshot.Name, volume.Name, string(types.ProgressStateInProgress), progress, backup, ""); updateErr != nil {
+				logrus.WithError(updateErr).Error("Failed to update backup status")
```
Comment (innobead) [inline deltablock.go:301]: It seems no hurt and not critical if encountering this error. Should we just print it in warn level?

File: fsops/fsops.go [hunk 3]
```diff
@@ -127,7 +127,9 @@ func (f *FileSystemOperator) List(path string) ([]string, error) {
 func (f *FileSystemOperator) Upload(src, dst string) error {
 	tmpDst := dst + ".tmp" + "." + strconv.FormatInt(time.Now().UTC().UnixNano(), 10)
 	if f.FileExists(tmpDst) {
-		f.Remove(tmpDst)
+		if err := f.Remove(tmpDst); err != nil {
+			return err
```
Comment (innobead) [inline fsops/fsops.go:131]: Should we handle this error?

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. Just a few comments. 

Just one feedback about when dealing an error but it's not critical and unhandled or it can be recovered eventually, suggest logging with warn level instead of error.

File: fsops/fsops.go (hunk 3 above)
Comment (FrankYang0529) [reply fsops/fsops.go:131]: Yeah, I think we return error and for the caller to handle it. Or do you mean log it before return the error?

**Comment by FrankYang0529** [comment]
> Just one feedback about when dealing an error but it's not critical and unhandled or it can be recovered eventually, suggest logging with warn level instead of error.

Thank you. I change the log level to warning.

File: fsops/fsops.go (hunk 3 above)
Comment (innobead) [reply fsops/fsops.go:131]: In the experience, usually removing files would be best efforts, so the error could not be handled. Or, we can just have a warn log, good enough? (of course, those temp files could be leftover.)

**Pull request merged**